controllers/check_instances_to_remove.go

```diff
@@ -34,29 +35,53 @@ type CheckInstancesToRemove struct{}
 
 // Reconcile runs the reconciler's work.
 func (c CheckInstancesToRemove) Reconcile(r *FoundationDBClusterReconciler, context ctx.Context, cluster *fdbtypes.FoundationDBCluster) (bool, error) {
-	hasNewRemovals := false
-
 	var removals = cluster.Status.PendingRemovals
 
 	if removals == nil {
 		removals = make(map[string]fdbtypes.PendingRemovalState)
 	}
 
 	for _, instanceID := range cluster.Spec.InstancesToRemove {
+		_, present := removals[instanceID]
+		if !present {
+			removals[instanceID] = fdbtypes.PendingRemovalState{}
+		}
+	}
+
+	for _, instanceID := range cluster.Spec.InstancesToRemoveWithoutExclusion {
+		_, present := removals[instanceID]
+		removalState := fdbtypes.PendingRemovalState{}
+		if !present {
```

Comment:
You're checking if a value isn't present, then fetching it if it's not?

Reply:
You're right, this is backward. It didn't come up in my testing because the data can be reconstructed most of the time. I think it can be made much simpler now.